src/pages/Login/useLoginForm.ts

```diff
@@ -0,0 +1,87 @@
+import { useState, useMemo } from 'react';
+import { useNavigate, useLocation } from 'react-router-dom';
+
+const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+export const useLoginForm = () => {
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState('');
+
+  const [password, setPassword] = useState('');
+  const [passwordError, setPasswordError] = useState('');
+
+  const navigate = useNavigate();
+  const location = useLocation();
+  const from = location.state?.from?.pathname || '/';
+
+  // 이메일 변경 핸들러
+  const handleEmailChange = (value: string) => {
+    setEmail(value);
+
+    if (!value) {
+      setEmailError('ID를 입력해주세요.');
+    } else if (!EMAIL_REGEX.test(value)) {
+      setEmailError('ID는 이메일 형식으로 입력해주세요.');
+    } else {
+      setEmailError('');
+    }
+  };
```

Comment:
```suggestion
  // 파생 상태: emailError는 email 값에 따라 자동 계산
  const emailError = useMemo(() => {
    if (!email) return 'ID를 입력해주세요.';
    if (!EMAIL_REGEX.test(email)) return 'ID는 이메일 형식으로 입력해주세요.';
    return '';
  }, [email]);
  
  const handleEmailChange = (value: string) => {
    setEmail(value);
  }
```

error를 상태로 관리하지 않아도 무방하답니다!

Reply:
상태로 관리하던 에러 메시지를 email/password 값에서 파생되는 값으로 계산하도록 리팩터링했습니다. 
구조는 유지하면서도 코드가 더 간결해졌습니다. 좋은 피드백 감사합니다!